src/pick.c

```diff
@@ -463,6 +470,7 @@ selected_choice(void)
 			filter_choices();
 			selection = 0;
 		}
+		yscroll = 0;
```

Comment:
As far as I understand, we only want to do this when the query is changed or put differently; whenever we do the following:

```
filter_choices();
selection = 0;
```

Reading through the code, I notice this bit being duplicated in all cases where the query is changed.

In all other cases we also want to skip resetting `yscroll`. For example, in the left and right arrow key cases. Currently after advancing `yscroll`, pressing left or right arrow will reset it to `0`.

I see two possible solutions to this:

- We could add `yscroll = 0;` wherever the above bit is duplicated. This would introduce more duplication but would fix the bug.
- Alternatively, we could extract the call to `filter_choices` and resetting the `selection` to just above this line and add `continue` statements to all branches that shouldn't go through the reset.